modules/dag-l1/src/main/scala/io/constellationnetwork/dag/l1/domain/transaction/TransactionService.scala

```diff
@@ -38,11 +40,37 @@ object TransactionService {
     SI <: SnapshotInfo[P]
   ](
     transactionStorage: TransactionStorage[F],
-    lastSnapshotStorage: LastSnapshotStorage[F, S, SI],
+    lastSnapshotStorage: LastSnapshotStorage[F, S, SI] with LatestBalances[F],
     transactionValidator: TransactionValidator[F],
-    mptStore: MptStore[F, GlobalStateKey]
+    mptStore: MptStore[F, GlobalStateKey],
+    shouldUseMptStore: Boolean
```

Comment:
what is the problem that prompted this flag? If it was for reading I could understand but the `useGlobalSnapshotInfo` method has a `tryPut` which would modify the GSI right? Then couldn't the two sources of state get out of sync?